mempool/src/lib.rs

```diff
@@ -21,17 +21,19 @@ use chainstate::chainstate_interface::ChainstateInterface;
 use common::chain::{Block, ChainConfig};
 use common::primitives::{BlockHeight, Id};
 use common::time_getter::TimeGetter;
-use pool::MempoolInterface;
+pub use interface::MempoolInterface;
 
-use crate::config::GetMemoryUsage;
 use crate::error::Error as MempoolError;
-use crate::pool::Mempool;
-use crate::pool::MempoolInterfaceHandle;
+use crate::mempool_interface_impl::MempoolInterfaceImpl;
+use get_memory_usage::GetMemoryUsage;
+
+pub use crate::mempool_interface_impl::SystemUsageEstimator;
```

Comment:
Please create a module called interface, and put the interface and impl in it. Also implement the calls for Deref and DerefMut. See how it's done in chainstate.

Reply:
Done.